Sources/Basics/Archiver/ZipArchiver.swift

```diff
@@ -90,32 +99,108 @@ public struct ZipArchiver: Archiver, Cancellable {
                 arguments: ["tar.exe", "-a", "-c", "-f", destinationPath.pathString, directory.basename],
                 workingDirectory: directory.parentDirectory.underlying
             )
+            try self.launchAndWait(process: process, completion: completion)
+            #elseif os(Linux)
+            // This is to work around `swift package-registry publish` tool failing on
+            // Amazon Linux 2 due to it having an earlier Glibc version (rdar://116370323)
+            // and therefore posix_spawn_file_actions_addchdir_np is unavailable.
+            // Instead of TSC.Process, we shell out to Foundation.Process and do `cd`
+            // explicitly before `zip`.
+            if SPM_posix_spawn_file_actions_addchdir_np_supported() {
+                try self.compress_zip(
+                    directory: directory,
+                    destinationPath: destinationPath,
+                    completion: completion
+                )
+            } else {
+                let process = Foundation.Process()
```

Comment:
we should probably use TSC's process since it handles stdout and stderr better

Reply:
If we use `TSC.Process` we would hit the same blocker (i.e., the Glibc check) that's preventing `publish` from working.

Reply:
only if we set the working directory, no?

Reply:
in fact I think if you set the working directory in Foundation process we will also run into the same issue

Reply:
Using `posix_spawn_file_actions_addchdir` is _good_, that's what we want right? The issue is just if that's not available, then we need to fail (and not just call `chdir`).

Reply:
We want to make this work when `posix_spawn_file_actions_addchdir` is not available instead of failing. And rather than having different code paths, we do the same for all scenarios (except on Windows) so we'd know sooner if it breaks.

Reply:
@yim-lee the _only_ way to make this work on UNIX without `posix_spawn_file_actions_addchdir` is to reimplement with fork+exec.

Do you happen to know which systems lack `posix_spawn_file_actions_addchdir`?

Reply:
@weissi We are trying to fix this for Amazon Linux 2, which has an earlier version of glibc.

Reply:
okay, thanks. if that's still an important target platform then it would need to be fork+exec there